### PR TITLE
genfit: Add latest tag 02-00-05

### PR DESCRIPTION
**Pull request by tmadlener**
Add the latest tags for genfit and update the changed root dependency / conflict for all but the latest tag.

### CODE REVIEW

**Review by mirguest — APPROVED** [review]

File: var/spack/repos/builtin/packages/genfit/package.py [hunk 1]
```diff
@@ -29,14 +33,18 @@ class Genfit(CMakePackage):
...
-        root_prefix = self.spec["root"].prefix
-        args.append("-DROOT_DIR=%s" % root_prefix)
+        if self.spec.satisfies("@:02-00-04"):
+            # normally, as a cmake package root should be
+            # automatically picked up after 'depends_on'
+            # as it is added to CMAKE_PREFIX_PATH
+            # but genfit cooks its own root cmake config
+            # so this workaround is needed for now.
+            root_prefix = self.spec["root"].prefix
+            args.append("-DROOT_DIR=%s" % root_prefix)
```
Comment (tldahlgren) [inline var/spack/repos/builtin/packages/genfit/package.py:48]: Nit.  You could use a python f-string here (see https://docs.python.org/3/tutorial/inputoutput.html).

**Review by tldahlgren — APPROVED** [review]
Confirmed the four version sha256.  I recommend but won't require use of the python f-string for formatting.  

~@mirguest do you want to review this PR before it is merged?~ (I now see you already did.)

**Comment by tldahlgren** [comment]
I'm going to go ahead and merge as is but recommend looking at using f-strings for formatting where it can be done.  We have been doing that when making changes to the core software.

**Comment by tmadlener** [comment]
Thanks :) I will keep the f-string changes in mind for the future. I wasn't aware of the recommendation yet.

**Pull request merged**